vmngclient/api/task_status_api.py

```diff
@@ -0,0 +1,123 @@
+import logging
+from typing import List, cast
+
+from tenacity import retry, retry_if_result, stop_after_attempt, wait_fixed  # type: ignore
+
+from vmngclient.session import vManageSession
+from vmngclient.utils.creation_tools import get_logger_name
+from vmngclient.utils.operation_status import OperationStatus, OperationStatusId
+
+logger = logging.getLogger(get_logger_name(__name__))
+
+
+class TaskStatus:
+    """
+    API class to check task status
+        Example usage:
+
+        session = create_vManageSession(ip_address,admin_username,password,port=port)
+        devices = DevicesAPI(session).devices
+        vsmart_device = [dev for dev in devices if dev.personality == Personality.VSMART][0]
+        exit_statuses = [OperationStatus.SUCCESS.value, OperationStatus.FAILURE.value]
+        exit_statuses_ids = [OperationStatusId.SUCCESS.value, OperationStatusId.FAILURE.value]
+
+        reboot_action = RebootAction(session,devices)
+        reboot_action.execute()
+
+        # Keep asking for reboot status until it's not in exit_statuses (Failure or Success)
+        # or timeout is not achieved (3000s)
+        TaskStatus(session).wait_for_completed(5,3000,exit_statuses,exit_statuses_ids,reboot_action.action_id)
+
+    """
+
+    def __init__(self, session: vManageSession) -> None:
+        self.session = session
+        self.status: str = ''
+        self.status_id: str = ''
+        self.activity: List[str] = []
+
+    def wait_for_completed(
+        self,
+        sleep_seconds: int,
+        timeout_seconds: int,
+        action_id: str,
+        exit_statuses: List[OperationStatus] = [
```

Comment:
can be simplified when one enum will be used with string pairs (preferably a dataclass)